docs/developer/go.md

```diff
@@ -1,3 +1,66 @@
 # Go
 
 An example application that uses the `bpfd-go` bindings can be found [here](https://github.com/redhat-et/bpfd/tree/main/examples/gocounter)
+
+## Prerequisites
+
+**Additive to the the ones defined in the [main README.md](../../README.md)**
+
+1. golang >= 1.18
+
+2. libbpf development package to get the required bpf c headers
+
+    Fedora:
+
+    `sudo dnf install libbpf-devel`
+
+    Ubuntu:
+
+    `sudo apt-get install libbpf-dev`
+
+3. Cilium's `bpf2go` binary
+
+    `go install github.com/cilium/ebpf/cmd/bpf2go@master`
+
+## Building
+
+To rebuild the c based bpf counter program example run:
+
+    pushd examples/gocounter && go generate && popd
+
+To build the userspace go client run:
+    
+    pushd examples/gocounter && go build && popd
+
+## Running 
+
+First start or ensure `bpfd` is up and running. 
+
+Then start the go program with:
+    pushd examples/gocounter && sudo ./gocounter <INTERNET INTERFACE NAME> && popd
```

Comment:
Nit. Should use backticks to make this a code block

Reply:
Done